src/mqtt_client/mod.rs

```diff
@@ -1,3 +1,19 @@
 mod messages;
 mod mqtt_client;
 pub use mqtt_client::MqttClient;
+
+#[non_exhaustive]
+pub enum HandlerResult {
+    /// A settings updated was accepted.
+    UpdateAccepted,
+
+    /// The application of a setting had ramifications on other settings values. In this case, all
+    /// settings will be republished.
+    UpdateSideEffects,
+}
+
+impl From<&()> for HandlerResult {
+    fn from(_: &()) -> Self {
+        HandlerResult::UpdateAccepted
+    }
+}
```

Comment:
I thought about it a bit more. I think we can skip this. Instead have the `Result::Ok` from the handler be returned from `update() -> Result` to the application. Then the application can decide based on that result if and what to do. We just need to make sure we expose the republication API to the application.
What do you think?

Reply:
And we should also give the handler two more bits of information if possible: a ref to the old settings and the path of the update.

Reply:
I've added additional info to the handler.

However, we can't allow the `update()` to return the result of the handler, since the handler may be called N times due to the MQTT poll loop. We can refactor that in the future, but for now, I've updated it so that the handler has to return Result<(), E>`

Reply:
Can't we just exit the loop on error? The unhandled incoming data will still be there (and presumably be handled next time) and the TX buffer is guaranteed to drain (so that there will be no deadlock).

Reply:
As of right now, minimq doesn't provide a mechanism to return an `Error` from inside the closure - the result of processing is unused when considering continuing iteration:
https://github.com/quartiq/minimq/blob/e8a072726ca94efbd2048917218cb2ba24b5c599/src/mqtt_client.rs#L413

I'm not opposed to refactoring minimq to support this behavior - I'll open up an issue.

Reply:
Ack. I'll leave it to you whether you want to wait for that in `minimq` and then follow up here or land this here first and then refactor later. But it'll ripple through the stack.

Reply:
Actually, after testing it out more, the application can just pass information back via accessing external state in the closure, so it can track any results that way if need be. I actually think the `Result` of this closure should be used for transferring a response back to the original requestee.

For example:
```rust
let mut republish_required = false;
client.handled_update(|path, old, new| {
    if new.threshold > MAX_THRESHOLD {
        return Err("Requested threshold above maximum";
    }

    *old = new.clone();
    Ok(())
}).unwrap()
```